x-pack/plugins/fleet/server/services/security/message_signing_service.ts

```diff
@@ -135,23 +135,38 @@ export class MessageSigningService implements MessageSigningServiceInterface {
     return publicKey;
   }
 
-  public async rotateKeyPair(): Promise<boolean> {
-    const isRemoved = await this.removeKeyPair();
-    if (isRemoved) {
-      await this.generateKeyPair();
-      return true;
+  public async rotateKeyPair(): Promise<{ error?: Error; success?: true }> {
```

Comment:
nit: is there any value to return a boolean, should just the method throw if there is an error?

Reply:
ohhh.. .disregard my comment above. I see you all agreed on `throw` :)